pkg/common/cns-lib/vsphere/virtualmachine_test.go

```diff
@@ -0,0 +1,204 @@
+	defer s.Close()
+
+	// Create a new client
+	client, err := govmomi.NewClient(ctx, s.URL, true)
+	if err != nil {
+		t.Fatalf("Failed to create new govmomi client, err: %v\n", err)
+	}
+	defer func() {
+		err := client.Logout(ctx)
+		if err != nil {
+			t.Logf("failed to logout govmomi client, err: %v", err)
+		}
+	}()
+
+	userinfo := simulator.DefaultLogin
+	// Create a new rest client and a tag manager
+	restClient := rest.NewClient(client.Client)
+	err = restClient.Login(ctx, userinfo)
+	if err != nil {
+		t.Fatalf("failed to login to the rest client, err: %v", err)
+	}
+	defer func() {
+		err := restClient.Logout(ctx)
+		if err != nil {
+			t.Logf("failed to logout rest client, err: %v", err)
+		}
+	}()
+
+	tagManager := tags.NewManager(restClient)
+	if tagManager == nil {
+		t.Fatalf("failed to create a new tagManager")
+	}
+	fmt.Printf("New tag manager created with useragent %s\n", tagManager.UserAgent)
+
+	// Create a new region category
+	regionCategoryId, err := CreateNewCategory(ctx, regionCategoryName, "SINGLE", tagManager)
+	if err != nil {
+		t.Fatalf("Error creating a new category %s, err: %v\n", regionCategoryName, err)
+	}
+	// Create a new tag within the region category
+	description := "region-tag"
+	regionTagId, err := CreateNewTag(ctx, regionCategoryId, regionTagName, description, tagManager)
+	if err != nil {
+		t.Fatalf("Error creating a new tag %s, err: %v\n", regionTagName, err)
+	}
+
+	// Add region tag to the datacenter
+	dcMoRef := simulator.Map.Any("Datacenter").(*simulator.Datacenter).Reference()
+	err = AttachTag(ctx, regionTagId, dcMoRef, tagManager)
+	if err != nil {
+		t.Fatalf("Error adding region tag to datacenter, err: %v", err)
+	}
+
+	// Create a new zone category
+	zoneCategoryId, err := CreateNewCategory(ctx, zoneCategoryName, "SINGLE", tagManager)
+	if err != nil {
+		t.Fatalf("Error creating a new category %s, err: %v\n", zoneCategoryName, err)
+	}
+	// Create a new tag within the zone category
+	description = "zone-tag"
+	zoneTagId, err := CreateNewTag(ctx, zoneCategoryId, zoneTagName, description, tagManager)
+	if err != nil {
+		t.Fatalf("Error creating a new tag %s, err: %v\n", zoneTagName, err)
+	}
+
+	// Add zone tag to the cluster
+	clusterMoRef := simulator.Map.Any("ClusterComputeResource").(*simulator.ClusterComputeResource).Reference()
+	err = AttachTag(ctx, zoneTagId, clusterMoRef, tagManager)
+	if err != nil {
+		t.Fatalf("Error adding region tag to datacenter, err: %v", err)
+	}
+
+	// Get any nodeVM instance which belongs to above datacenter and cluster and check GetZoneRegion of that VM
+	dcobj := object.NewDatacenter(client.Client, dcMoRef)
+	dc := &Datacenter{
+		Datacenter: dcobj,
+	}
+	obj := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+	vm := &VirtualMachine{
+		Datacenter:     dc,
+		VirtualMachine: object.NewVirtualMachine(client.Client, obj.Reference()),
+	}
+
+	zone, region, err := vm.GetZoneRegion(ctx, zoneCategoryName, regionCategoryName, tagManager)
```

Comment:
Can you explain what GetZoneRegion does? the params don't make it easy to understand

Reply:
Added a comment